File: SaliScoreDesktop/score/CsCompositionHeader.cpp
#include "CsCompositionHeader.h"

CsCompositionHeader::CsCompositionHeader()
  {

  }




void CsCompositionHeader::clear()
  {
  CsDescrSong::clear();
  mSettings.clear();
  }




void CsCompositionHeader::jsonWrite(CsJsonWriter &js) const
  {
  CsDescrSong::jsonWrite( js );
  mSettings.jsonWrite( js );
  js.jsonObject<CsTrainInterval>( "fragment0", mFragments[0] );
  js.jsonObject<CsTrainInterval>( "fragment1", mFragments[1] );
  }




void CsCompositionHeader::jsonRead(CsJsonReader &js)
  {
  CsDescrSong::jsonRead( js );
  mSettings.jsonRead( js );
  js.jsonObject<CsTrainInterval>( "fragment0", mFragments[0] );
  js.jsonObject<CsTrainInterval>( "fragment1", mFragments[1] );
  }

File: SaliScoreAndroid/app/src/main/java/salilab/saliscore/CsCompositionHeader.java
package salilab.saliscore;

public class CsCompositionHeader extends CsDescrSong {
  public CsSynthSettings mSettings;
  public CsTrainInterval mFragments[];

  CsCompositionHeader() {
    super();
    mSettings = new CsSynthSettings();
    mFragments = new CsTrainInterval[2];
    mFragments[0] = new CsTrainInterval();
    mFragments[1] = new CsTrainInterval();
  }

  CsCompositionHeader( CsCompositionHeader hdr ) {
    super( hdr );
    mSettings = new CsSynthSettings(hdr.mSettings);
    mFragments = new CsTrainInterval[2];
    mFragments[0] = new CsTrainInterval(hdr.mFragments[0]);
    mFragments[1] = new CsTrainInterval(hdr.mFragments[1]);
  }

  @Override
  void clear() {
    super.clear();
    mSettings.clear();
  }

  @Override
  void jsonWrite( SvJsonWriter js ) throws Exception {
    super.jsonWrite( js );
    mSettings.jsonWrite( js );
    js.jsonObject( "fragment0", mFragments[0] );
    js.jsonObject( "fragment1", mFragments[1] );
  }

  @Override
  void jsonRead( SvJsonReader js ) throws Exception {
    super.jsonRead( js );
    mSettings.jsonRead( js );
    js.jsonObject( "fragment0", mFragments[0] );
    js.jsonObject( "fragment1", mFragments[1] );
  }

  @Override
  CsJsonWritable copy() {
    return new CsCompositionHeader( this );
  }

  }
